src/machine/machine.c

```diff
@@ -88,20 +88,23 @@ Machine* machine_free(Machine *m) {
         while (m->operations)
                 operation_free(m->operations);
 
-        if (m->in_gc_queue)
+        if (m->in_gc_queue && m->manager)
```

Comment:
this seems unnecessary? to be in the gc queue the machine really must have been registered properly in the manager. would prefer to do this instead:

```c
if (m->in_gc_queue) {
        assert(m->manager);
        LIST_REMOVE(…);
}
```

because if the above doesn't hold we are in a state of memory corruption

Reply:
I wasn't sure if it was okay to use an assert in a cleanup function but I'll change it to this now.